kernel/main.c

```diff
@@ -200,10 +200,11 @@ read_multiboot_info(void)
       const char *name = TO_PTR(mbi->boot_loader_name);
 
       if (!strcmp(name, "TILCK_EFI")) {
+         tilck_extra_boot_info* extra_boot_info = (tilck_extra_boot_info*)mbi->apm_table;
```

Comment:
This will make the 64-bit build to fail, because `apm_table` is 32-bit. We cannot directly cast a 32-bit integer into a pointer. This needs a double cast. So, just use `TO_PTR(mbi->apm_table)`.

Also, coding style: in Tilck the "*" goes next to the variable, not next to the type as in modern C++. The reason? If you declare `int* a, b;` you'll have just `a` as pointer, not `b`.